src/runtime_src/core/pcie/driver/linux/xocl/subdev/feature_rom.c

```diff
@@ -578,23 +578,24 @@ static int get_header_from_peer(struct feature_rom *rom)
 	struct FeatureRomHeader *header;
 	struct resource res;
 	xdev_handle_t xdev = xocl_get_xdev(rom->pdev);
-	int ret;
+	int offset = 0;
+	const u64 *io_off;
 
 	header = XOCL_GET_SUBDEV_PRIV(&rom->pdev->dev);
 	if (!header)
 		return -ENODEV;
 
 	memcpy(&rom->header, header, sizeof(*header));
 
-	xocl_xdev_info(xdev, "Searching CMC in dtb.");
-	ret = xocl_subdev_get_resource(xdev, RESNAME_KDMA,
-			IORESOURCE_MEM, &res);
-	if (!ret) {
+	xocl_xdev_info(xdev, "Searching CDMA in dtb.");
+	offset = xocl_fdt_path_offset(xdev, XDEV(xdev)->fdt_blob, "/" NODE_ENDPOINTS "/" RESNAME_KDMA);
+	if (offset) {
+		io_off = xocl_fdt_getprop(xdev, XDEV(xdev)->fdt_blob, offset, PROP_IO_OFFSET, NULL);
```

Comment:
Minor Nit ....might be good to add error check for return value of xocl_fdt_getprop() just incase this node does not have IO offset property.

Reply:
Done.